### ISSUE TITLE
Maven Installation ERROR due to wrong Selenium WebDriver.class version.

### ISSUE THREAD

**Issue by SamBerl84** [MuleSoft-AI-Chain-Project/mule-web-crawler-connector#15]
While doing local maven installation for Web-crawler connector, I am getting below error. I am installing the connect with Java8. Please, share a fixture or a workaround.

**[ERROR] diagnostic: C:\Users\Sam\mac-web-crawler\src\main\java\com\mule\mulechain\crawler\internal\helpers\crawlingHelper.java:9: error: cannot access org.openqa.selenium.WebDriver
import org.openqa.selenium.WebDriver;
                          ^
  bad class file: C:\Repo\org\seleniumhq\selenium\selenium-api\4.26.0\selenium-api-4.26.0.jar(org/openqa/selenium/WebDriver.class)
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.**

![image](https://github.com/user-attachments/assets/3e726e28-dd2c-4a24-9d85-9b30ba5e9423)

Regards,
Sameer

**Comment by amirkhan-ak-sf** [comment]
Try it with another Java Version > 8, i.e. 11 or 17.

Perform the following 2 steps:
`export MAVEN_OPTS="--add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.util.regex=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED"`

And based on the java Version run one of the following command

` 
For Java 17
mvn clean install -Dmaven.test.skip=true -DskipTests -Djdeps.multiRelease=17
 
For Java 21
mvn clean install -Dmaven.test.skip=true -DskipTests -Djdeps.multiRelease=21
 
For Java 22
mvn clean install -Dmaven.test.skip=true -DskipTests -Djdeps.multiRelease=22`

Make sure to set the JAVA_HOME accordingly to the java Version you have installed.


Hope this helps.

Cheers

**Comment by tbolis-at-mulesoft** [comment]
Yesterday I fixed the pom.xml that was causing issues during the build phase. Using java 11 you should not face the class file version issue. Can you share your studio jar file?

**Comment by ajaytiwa** [comment]
Sharing the Studio config..

Flow Config

<flow name="WebSiteCrawling" doc:id="07a1cf71-30a7-4040-9189-5e4a73a9c412" >
		<http:listener doc:name="Listener" doc:id="4f64b0b3-e8c6-4fe3-a332-cf2a68fdf216" config-ref="HTTP_Listener_config" path="/webcrawling"/>
		<mac-web-crawler:crawl-website doc:name="Crawl website" doc:id="eb3b9591-80be-4502-b2dd-55d1a0a82ccc" config-ref="MAC_WebCrawler_Config" url="<<here I have private URL>>" downloadPath='#["C:\\TEMP\\webcrawling\\"]' maxDepth="1" />
	</flow>
	

Connector Config

<mac-web-crawler:config name="MAC_WebCrawler_Config" doc:name="MAC WebCrawler Config" doc:id="ef750de6-d274-4a32-9fd6-b5e4b38567b3" >
	</mac-web-crawler:config>

**Comment by ajaytiwa** [comment]
Flow config:


<flow name="WebSiteCrawling" doc:id="07a1cf71-30a7-4040-9189-5e4a73a9c412" >
		<http:listener doc:name="Listener" doc:id="4f64b0b3-e8c6-4fe3-a332-cf2a68fdf216" config-ref="HTTP_Listener_config" path="/webcrawling"/>
		<mac-web-crawler:crawl-website doc:name="Crawl website" doc:id="eb3b9591-80be-4502-b2dd-55d1a0a82ccc" config-ref="MAC_WebCrawler_Config" url="<<website_url>>" downloadPath='#["C:\\TEMP\\webcrawling\\"]' maxDepth="1" />
	</flow>

**Comment by ajaytiwa** [comment]
@SamBerl84 Just make sure you have latest repo cloned..
I had the same issue but after cloning the latest repo from master branch(with java 11) , didn't encountered issue you mentioned recently, however I am still facing the error related to selenium version.

**Comment by SamBerl84** [comment]
@ajaytiwa 
Yes, tried again with latest repo, same error. 
Its with Java17, I guess I will give Java11 a go to check if that does something different.

Cheers

**Comment by amirkhan-ak-sf** [comment]
What is the command you used?  Can you share it here @SamBerl84 ?

**Comment by SamBerl84** [comment]
@amirkhan-ak-sf 

I was able to build it successfully (using Java11). Updated the dependency version in POM file as below.

      		**<version>0.1.25-SNAPSHOT</version>**

@ajaytiwa 
I also faced below error when making http calls from POSTMAN. This got resolved by updating Anypoint Studio (which updated Java Runtime).

**java.lang.UnsupportedClassVersionError: org/openqa/selenium/WebDriver has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0**


Thank you all for your support! 
Crawling Begins....!! =D
